src/stream/src/executor/mview/materialize.rs

```diff
@@ -58,6 +58,19 @@ pub struct MaterializeExecutor<S: StateStore, SD: ValueRowSerde> {
     conflict_behavior: ConflictBehavior,
 
     version_column_index: Option<u32>,
+
+    may_have_downstream: bool,
+}
+
+fn get_op_consistency_level(
+    conflict_behavior: ConflictBehavior,
+    may_have_downstream: bool,
+) -> StateTableOpConsistencyLevel {
+    if may_have_downstream || !matches!(conflict_behavior, ConflictBehavior::Overwrite) {
+        StateTableOpConsistencyLevel::ConsistentOldValue
+    } else {
+        StateTableOpConsistencyLevel::Inconsistent
+    }
```

Comment:
Can we use `&&` instead of `||` for the condition? which could be easier to understand. BTW, we can add this comment back `// Table with overwrite conflict behavior could disable conflict check if no downstream mv depends on it, so we use a inconsistent_op to skip sanity check as well.`